### DISCUSSION TITLE
DataTerminal input handling.

### DISCUSSION THREAD

**Discussion by rszemeti** [peterantypas/maiana#126] [General]
It seems to struggle if you paste stuff like station info into Putty or use a buffered terminal that only sends on cr/lf.. Maybe there is a bit too much going on in the callback handler.

Characters get missed etc.  so if you compile your station string in a text editor and paste it into Putty nothing happens (parsed as too few params) or if it does get through, bits of the callsign/mmsi/name are missing 

Fine if you type it char by char though.  I was using the serial terminal in Arduino which only sends when return is pressed and nothing was working.   Might be worth fixing or perhaps just a note not to cut/paste in Putty.

**Comment by peterantypas** [comment]
I generally advocate for terminal programs that use line mode like Winterm
or CoolTerm.
